src/test/java/jewellery/inventory/unit/service/ProductServiceTest.java

```diff
@@ -0,0 +1,264 @@
+
+    productService.deleteProduct(testProduct.getId());
+
+    assertEquals(0, productRepository.count());
+  }
+
+  @Test
+  void testDeleteProductShouldThrowExceptionWhenProductIsPartOfProduct() {
+    testProduct.setContentOf(new Product());
+    when(productRepository.findById(testProduct.getId())).thenReturn(Optional.of(testProduct));
+    UUID productId = testProduct.getId();
+    assertThrows(ProductIsContentException.class, () -> productService.deleteProduct(productId));
+  }
+
+  @Test
+  void testDeleteProductShouldThrowExceptionWhenProductIsSold() {
+    testProduct.setSold(true);
+    when(productRepository.findById(testProduct.getId())).thenReturn(Optional.of(testProduct));
+    UUID productId = testProduct.getId();
+    assertThrows(ProductIsSoldException.class, () -> productService.deleteProduct(productId));
+  }
+
+  @Test
+  void testDeleteProductGetProductShouldThrowExceptionWhenProductNotExist() {
+    UUID fakeId = UUID.randomUUID();
+    assertThrows(ProductNotFoundException.class, () -> productService.deleteProduct(fakeId));
+  }
+
+  @Test
+  void createProductSuccessfully() {
+
+    when(userRepository.findById(productRequestDto.getOwnerId())).thenReturn(Optional.of(user));
+    when(resourceInUserRepository.findByResourceIdAndOwnerId(pearl.getId(), user.getId()))
+        .thenReturn(Optional.of(resourceInUser));
+    user.setResourcesOwned(List.of(resourceInUser));
+
+    ProductResponseDto response = new ProductResponseDto();
+    when(productMapper.mapToProductResponseDto(any())).thenReturn(response);
+
+    ProductResponseDto actual = productService.createProduct(productRequestDto);
+
+    assertEquals(actual, response);
+  }
+
+  @Test
+  void testCreateProductShouldRemoveResourceFromUserWhenResourceInUserIsZero() {
+    when(userRepository.findById(user.getId())).thenReturn(Optional.of(user));
+    when(resourceInUserRepository.findByResourceIdAndOwnerId(pearl.getId(), user.getId()))
+        .thenReturn(Optional.of(resourceInUser));
+    resourceInUser.setQuantity(5);
+    user.setResourcesOwned(List.of(resourceInUser));
+    resourceQuantityRequestDto.setQuantity(5);
+
+    productService.createProduct(productRequestDto);
+
+    assertEquals(0, resourceInUserRepository.count());
+  }
+
+  @Test
+  void testCreateProductShouldThrowExceptionWhenResourceInUserNotFound() {
+    user.setResourcesOwned(new ArrayList<>());
+    when(userRepository.findById(user.getId())).thenReturn(Optional.of(user));
+
+    assertThrows(
+        ResourceInUserNotFoundException.class,
+        () -> productService.createProduct(productRequestDto));
+  }
+
+  @Test
+  void testCreateProductShouldThrowWhenResourceNotExistInResourceInUserRepository() {
+    when(userRepository.findById(user.getId())).thenReturn(Optional.of(user));
+    assertThrows(
+        ResourceInUserNotFoundException.class,
+        () -> productService.createProduct(productRequestDto));
+  }
```

Comment:
This should also be deleted? Is it different than the test above?

Reply:
Removed!